enhancements/sig-architecture/82-addon-template/README.md

```diff
@@ -0,0 +1,344 @@
+# Add-on Template
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+This proposal adds a new CRD addonTemplate, to simplify the development of an add-on. it makes developers no longer
+need to compile and deploy an addon manager for each add-on.
+
+## Motivation
+
+Currently, OCM provides the addon framework, which is a library that provides functions such as addon installation and
+registration. Developers can use the addon framework to implement the corresponding interface, define the resources that
+need to be deployed on the managed cluster, and then build the addon image and deploy it to the hub cluster. After that,
+The addon can be enabled for the corresponding managed cluster by creating a `ManagedClusterAddon`.
+
+This simplifies the addon development process, but it still requires developers to implement the interface defined in
+the addon-framework via code, and the development process is still slightly complicated. so we hope to explore a more
+straightforward addon development way.
+
+### Goals
+
+- Provide a new addon development model, developers can define an addon YAML template and some parameters to describe
+  how the addon agent will be installed on the managed cluster and how to register it to the hub, developers do not need
+  to deploy an addon-manager on the hub cluster to manage the addon agent.
+- Make the addon agent deployment and registration configurable, modifying the deployment and registration logic does
+  not need to recompile the addon image.
+
+### Non-Goals
+
+- The addon agent image is still required.
```

Comment:
For complicated addon which has special rbac requirement etc would still need to use addon-framework